Remove redundant names and logging.

stuhood opened the pull request:
### Problem

Post #9894, it's clear that many named rules are redundant, because the processes that run below them have better descriptions (relates to #7907, although it's possible that we'd like to drive the names of a workunit in some other way).

Example for a `--no-process-execution-use-local-cache` run:
```
18:00:22:852 [INFO] Starting tests: tests/python/pants_test/util:strutil
18:00:22:870 [INFO] Completed: Create a PEX from targets
18:00:24:029 [INFO] Completed: Building test_runner.pex
18:00:24:030 [INFO] Completed: Building requirements.pex
18:00:24:031 [INFO] Completed: Create PEX
18:00:24:031 [INFO] Completed: Create PEX
18:00:32:638 [INFO] Completed: Building pytest.pex with 8 requirements: ipdb, pygments, <snip>
18:00:32:639 [INFO] Completed: Create PEX
18:00:34:480 [INFO] Completed: Run Pytest for tests/python/pants_test/util:strutil
18:00:34:480 [INFO] Completed: Run pytest
18:00:34:481 [INFO] Tests succeeded: tests/python/pants_test/util:strutil
✓ tests/python/pants_test/util:strutil
============================= test session starts ==============================
<snip>
============================== 6 passed in 0.03s ===============================

18:00:34:481 [INFO] Completed: `test` goal
```

Likewise, when the process cache is hit on a run without `pantsd`, the `@rules` will run ("Create PEX" etc) but the processes won't because they'll have been persistently cached... which makes the output even less useful.

### Solution

While we should pursue #7907, and likely also allow `@rule`s to explicitly declare their level, some rule names just won't be useful when compared to the process names we've generated. Prune a few.

### Result

Output for the same uncached command is:
```
00:35:51:820 [INFO] Completed: Building requirements.pex
00:35:51:821 [INFO] Completed: Building test_runner.pex
00:36:02:208 [INFO] Completed: Building pytest.pex with 8 requirements: ipdb, pygments, <snip>
00:36:04:487 [INFO] Completed: Run Pytest for tests/python/pants_test/util:strutil
00:36:04:488 [INFO] Tests succeeded: tests/python/pants_test/util:strutil
✓ tests/python/pants_test/util:strutil
============================= test session starts ==============================
<snip>
============================== 6 passed in 0.03s ===============================

18:05:23:318 [INFO] Completed: `test` goal
```

See also: the travis output for this run.

[ci skip-jvm-tests]
[ci skip-rust-tests]

stuhood commented:
> I agree with the `Create PEX` change and "started test" change.

FTR: "Started: $workunit" will still render if the UI is disabled: so we'll be able to see tests starting, but more accurately than before, because it will only be when (and if) the process actually starts.

> One downside of this PR, though, is that the test logging will no longer say "Succeeded or Failed". That has been very helpful in the past when CI, for example, times out, so you don't get the actual results printed and you don't know what failed or passed.

Yea, we'll need to fix that soon: there is a sketch of how to do it by allowing return types to change the level of the workunit as it completes.

> The best solution is to have streaming workunits that return the result right away. In the short term, I don't know what the solution is? I agree it's redundant to both have the workunit log and our own log, but we're losing information now.

The rendering of workunits completeing is streaming... not sure what you mean. See the recordings on #9894.

> Also, FYI we use logging with `binary` and `setup-py`. I don't know how we should treat those. It's also important that the log message says exactly where we outputted the file. And we want that log to happen regardless of `--no-dynamic-ui` or not.

Those will continue to render as log messages just as they do today.

stuhood commented:
Mm. In addition to the above, I just noticed that we now also log (one line: `04:30:38:798 [INFO] Completed: `list` goal`) for `./v2 list`. It seems that even goals should likely be able to have their workunits at `debug`.

gshuflin commented:
I agree with Eric - until we have a way to report individual test success/failure in a way linked to the workunit, we shouldn't delete those succeeded/failed messages (and building that infrastructure is something that we should work on in the very near future).

stuhood commented:
Fine with that. Added back the success/failure logging. But to cut down from 3 "Completed" messages per test, I also removed the name from `Run pytest`: see updated description.

benjyw commented:
While we're bikeshedding, do we need that final "Completed: `test` goal" message?

stuhood commented:
> While we're bikeshedding, do we need that final "Completed: `test` goal" message?

In general: no, I don't think we do. See https://github.com/pantsbuild/pants/pull/9921#issuecomment-636419552 as well. We could move goals to `debug` level, but that would be a bit strange for some consumers I think.

Deferring for a followup.

Eric-Arellano commented:
Thanks, Stu. This is certainly a step in the right direction.
